### ISSUE TITLE
Breaking changes from Leaflet 2.0/Esri Leaflet 2.0

### ISSUE THREAD

**Issue by patrickarlt** [Esri/esri-leaflet-renderers#62]
@jgravois @kneemer I've been going through the changes to get Esri Leaflet ready for ~~1.0~~ 2.0 and I've made some API changes in Esri Leaflet.

The biggest change is the removal of the special handling for `L.CircleMarker` that @jgravois added for the last release. This ended up causing a HUGE number of problems. After a pretty close reading of the Leaflet code we should do the following if we want to use markers based on `L.Path` like `L.CircleMarker`.

```js
L.esri.featureLayer('http://gis.example.com/mock/arcgis/rest/services/MockService/MockFeatureServer/0', {
      pointToLayer: function(feature, latlng){
        return L.circleMarker(latlng);
      },
      style: function () {
        return {
          color: 'green'
        };
      }
    }).addTo(map);
```

This code actually works in the current version of Esri Leaflet. `pointToLayer` should be used ONLY to convert the latlng into some kind of marker object. If you are using `L.Marker` you can style it here but for anything that inherits from `L.Path` you should style it using the `style` option.

**Comment by patrickarlt** [comment]
This is for 2.0 now. Not sure if anything will break in 1.0. You might just need a `return` if there is an error getting the metadata.

**Comment by jgravois** [comment]
pretty sure this is resolved by 4a32a446781270173065fd0bcaded23e20715b47
